.github/workflows/build.yml

```
@@ -0,0 +1,51 @@
+name: Website build
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: nodeversion
+        run: echo ::set-output name=NODE::$(cat .nvmrc)
+      - name: restore Cache
+        uses: actions/cache@master
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+      - name: Use Node.js ${{steps.nodeversion.outputs.NODE}}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.nodeversion.outputs.NODE }}
+      - name: install
+        run: |
+          yarn
+      - name: Lint
+        run: |
+          yarn run prettier:check
+          yarn run prettier:json:check
+          yarn run lint
+      - name: Cypress run
+        uses: cypress-io/github-action@v2
+        with:
+          start: yarn serve
+          wait-on: 'http://localhost:8080'
+      - name: build
+        run: |
+          echo VUE_APP_TRAVIS_COMMIT=$GITHUB_SHA >> .env.production
+          yarn run build
+          echo config.qmk.fm > dist/CNAME
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          deploy_key: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./dist
+          publish_branch: test-gh-pages
+          force_orphan: true
+          user_name: 'QMK Bot'
+          user_email: 'hello@qmk.fm'
+          commit_message: 'Deploy to GitHub Pages'
```

We currently use the following action everywhere else, so it would be good to align if possible.

Something like...
```suggestion
      uses: JamesIves/github-pages-deploy-action@3.7.1
      with:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        BRANCH: test-gh-pages
        FOLDER: ./dist
        GIT_CONFIG_NAME : QMK Bot
        GIT_CONFIG_EMAIL: hello@qmk.fm
```